src/Runner.Worker/ContainerOperationProvider.cs

```diff
@@ -299,15 +328,20 @@ private async Task StopContainerAsync(IExecutionContext executionContext, Contai
 
             if (!string.IsNullOrEmpty(container.ContainerId))
             {
-                if (!container.IsJobContainer)
+                if (!container.IsJobContainer && !container.FailedInitialization)
                 {
-                    // Print logs for service container jobs (not the "action" job itself b/c that's already logged).
-                    executionContext.Output($"Print service container logs: {container.ContainerDisplayName}");
-
-                    int logsExitCode = await _dockerManager.DockerLogs(executionContext, container.ContainerId);
-                    if (logsExitCode != 0)
+                    var healthcheck = await ContainerHealthcheck(executionContext, container);
```

Comment:
do we need to do this extra health check?
We already filter out using `if (!container.IsJobContainer && !container.FailedInitialization)`, I assume we should just print out logs for the rest containers?

Reply:
Right, fixed!